src/types.rs

```diff
@@ -281,165 +203,132 @@ impl BoundedStorable for PrincipalStorable {
     const IS_FIXED_SIZE: bool = false;
 }
 
-#[derive(Clone, Debug, Eq, PartialEq, CandidType, Deserialize)]
-pub struct ProviderView {
-    #[serde(rename = "providerId")]
-    pub provider_id: u64,
-    pub owner: Principal,
-    #[serde(rename = "chainId")]
-    pub chain_id: u64,
-    pub hostname: String,
-    #[serde(rename = "cyclesPerCall")]
-    pub cycles_per_call: u64,
-    #[serde(rename = "cyclesPerMessageByte")]
-    pub cycles_per_message_byte: u64,
-    pub primary: bool,
-}
+#[derive(Clone)]
```

Comment:
`Clone` is currently not used, so I would avoid introducing it for `ApiKey` (`Cloning` is as risky as `read`).

For additional safety, we could also use `#[derive(Zeroize, ZeroizeOnDrop)]` from this [crate](https://crates.io/crates/zeroize/1.8.1).

Reply:
Good point about `Clone`, since the API keys are no longer stored in `Provider`. Removed `Clone` and added the traits from `zeroize`.